### PR TITLE
(fix) Change page title of `channels` to be changed based on the enabled providers.

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
(fix) Change page title of `channels` to be changed based on the enabled providers.

### Related Issues
- https://github.com/wso2/product-is/issues/17805

### Screenshot
Screenshot 2023-11-19 at 12 35 30

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/notification-channels/pages/notification-channels.tsx [hunk 1]
```diff
@@ -51,6 +51,18 @@ export const NotificationChannelPage: FunctionComponent<NotificationChannelPageI
 
     const featureConfig : FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
 
+    /**
+     * Check if the email provider is enabled for the tenant.
+     */
+    const isEmailProviderEnabled: boolean = featureConfig.emailProviders?.enabled
```
Comment (pavinduLakshan) [inline apps/console/src/features/notification-channels/pages/notification-channels.tsx:57]:
```suggestion
    const isEmailProviderEnabled: boolean = featureConfig?.emailProviders?.enabled
```

File: apps/console/src/features/notification-channels/pages/notification-channels.tsx [hunk 2]
```diff
@@ -51,6 +51,18 @@ export const NotificationChannelPage: FunctionComponent<NotificationChannelPageI
...
+        && !(featureConfig?.emailProviders?.disabledFeatures?.includes("superTenantProvider")
+        && organizationType === OrganizationType.SUPER_ORGANIZATION);
+
+    /**
+     * Check if the SMS provider is enabled for the tenant.
+     */
+    const isSMSProviderEnabled: boolean = featureConfig.smsProviders?.enabled;
```
Comment (pavinduLakshan) [inline apps/console/src/features/notification-channels/pages/notification-channels.tsx:64]:
```suggestion
    const isSMSProviderEnabled: boolean = featureConfig?.smsProviders?.enabled;
```

File: apps/console/src/features/notification-channels/pages/notification-channels.tsx [hunk 3]
```diff
@@ -61,17 +73,44 @@ export const NotificationChannelPage: FunctionComponent<NotificationChannelPageI
...
+    
+    /**
+     * Get the page details based on the enabled providers.
+     */
+    const getPageDetails = (): {
+        description: string;
+        pageTitle: string;
+        title: string;
+    } => {
+        if (isEmailProviderEnabled === isSMSProviderEnabled) {
```
Comment (pavinduLakshan) [inline apps/console/src/features/notification-channels/pages/notification-channels.tsx:85]:
```suggestion
        if (isEmailProviderEnabled && isSMSProviderEnabled) {
```

**Review by DonOmalVindula — APPROVED** [review]

**Review by pavinduLakshan — DISMISSED** [review]

File: apps/console/src/features/notification-channels/pages/notification-channels.tsx (hunk 3 above)
Comment (Achintha444) [reply apps/console/src/features/notification-channels/pages/notification-channels.tsx:85]: Need to display this message, when the `email provider` and `sms provider` both are disabled. This is a hypothetical situation and placed as a fallback condition.

File: apps/console/src/features/notification-channels/pages/notification-channels.tsx (hunk 2 above)
Comment (Achintha444) [reply apps/console/src/features/notification-channels/pages/notification-channels.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4697/commits/19bc95f1bdae9247c8c41bcbb06c39994ff01f9f

File: apps/console/src/features/notification-channels/pages/notification-channels.tsx (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/notification-channels/pages/notification-channels.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4697/commits/19bc95f1bdae9247c8c41bcbb06c39994ff01f9f

File: apps/console/src/features/notification-channels/pages/notification-channels.tsx (hunk 3 above)
Comment (brionmario) [reply apps/console/src/features/notification-channels/pages/notification-channels.tsx:85]: Can't we go with ```if (!isEmailProviderEnabled && !isSMSProviderEnabled) {```

Comment (Achintha444) [reply apps/console/src/features/notification-channels/pages/notification-channels.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4697/commits/8bfeed8c422bd4ef5d39051766270113c17c5cc4

**Review by pavinduLakshan — APPROVED** [review]

**Review by DonOmalVindula — APPROVED** [review]

**Pull request merged**